test_files/test.rb

```diff
@@ -0,0 +1,71 @@
+require "net/http"
+
+# This is just your standard class
+#  nothing strange here
+class SomeClass
+  # In ruby, the parentheses can be omitted in many circumstances
+  # This is probably a job for the AST, but we'll add a test here anyway
+  def empty_parens()
+    "Nice"
+  end
+end
+
+module SomeModule
+  def omitted_parens
+    :foo
+  end
+end
+
+module Test ; end
+
+# This module is acting more like a namespace
+module Test::SubTest
```

Comment:
I see a few places with `::`. You may find adding this predicate relevant for those (already heavily used for CPP): https://github.com/Exafunction/codeium-parse/blob/7be8f1c9166acf0c9d544393ed895468e9999e29/queries/cpp_tags.scm#L11

Description of what this does from our docs:
```
#lineage-from-name!
    (#lineage-from-name! "literal")
    If the name captures scopes, split by "literal" and retain the last element
    as the name. The other elements are appended to the lineage.
```

Reply:
I've added it, but it doesn't seem to have altered the `.golden` file. Is this something that is caught elsewhere, and have I put it in the right place?

Reply:
Ah, so I see your scope captures but tree-sitter doesn't exactly let you capture an arbitrary number of them. I would propose you apply the following patch, and that should do it:
```patch
diff --git a/queries/ruby_tags.scm b/queries/ruby_tags.scm
index 43ad352..8c3c135 100644
--- a/queries/ruby_tags.scm
+++ b/queries/ruby_tags.scm
@@ -27,20 +27,8 @@
 (
   (comment)* @doc
   .
-  [
-    (class
-      name: [
-        (constant) @name
-        (scope_resolution
-          name: (_) @name)
-      ]) @definition.class
-    (singleton_class
-      value: [
-        (constant) @name
-        (scope_resolution
-          name: (_) @name)
-      ]) @definition.class
-  ]
+  (class
+    name: (_) @name) @definition.class
   (#lineage-from-name! "::")
   (#select-adjacent! @doc @definition.class)
 )
@@ -51,11 +39,7 @@
   (comment)* @doc
   .
   (module
-    name: [
-      (constant) @name
-      (scope_resolution
-        name: (_) @name)
-    ]) @definition.module
+    name: (_) @name) @definition.module
   (#lineage-from-name! "::")
   (#select-adjacent! @doc @definition.module)
 )
```